sympy/functions/special/gamma_functions.py

```diff
@@ -1092,11 +1094,8 @@ def _eval_conjugate(self):
 
     def _eval_is_real(self):
         x, p = self.args
-        if p.is_Integer:
-            for k in range(p):
-                num = (x - k/S(2))
-                if intlike(num) and num <= 0:
-                    return
-                elif num.is_nonpositive and num.is_integer:
-                    return
+        if x.is_Number and p.is_Number:
```

Comment:
Why this condition?
If you leave it out and account for https://github.com/sympy/sympy/pull/16843#discussion_r291815199, it could allow symbolic expressions as well.